src/challenges.js

```diff
@@ -30,16 +56,48 @@ const cacheFunction = cb => {
 const reverseStr = str => {
   // reverse str takes in a string and returns that string in reversed order
   // The only difference between the way you've solved this before and now is that you need to do it recursivley!
+  if (str === "") 
+      return "";
+    
+    else
+      return reverseStr(str.substr(1)) + str.charAt(0);
+    return  result;
 };
 
 const checkMatchingLeaves = obj => {
   // return true if every property on `obj` is the same
   // otherwise return false
+  const leaves = [];
+  const findLeaves = object => {
+    Object.values(object).forEach(value => {
+      if (value === Object(value)) {
```

Comment:
To check if `value` is an Object, we can do `typeof value === 'object'` instead.